go/proxyd/rpc.go

```diff
@@ -152,3 +164,52 @@ func IsBatch(raw []byte) bool {
 	}
 	return false
 }
+
+func isArchiveRequest(req *RPCReq, getBlockNum GetLatestBlockNumFn) (bool, error) {
+	const BLOCK_NUM_ARCHIVE_HEIGHT = 64
+
+	var (
+		tag            string
+		latestBlockNum uint64
+		err            error
+	)
+	switch req.Method {
+	case "eth_call", "eth_getBalance", "eth_getTransactionCount", "eth_getStorageAt", "eth_getCode":
+		tag, err = extractTerminalBlockTag(req)
+	case "eth_getBlockByNumber":
+		tag, _, err = decodeGetBlockByNumberParams(req.Params)
+	default:
+		return false, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+	if tag == "" || isBlockDependentParam(tag) {
+		return false, nil
+	}
+	// NOTE: we're reading from an in-memory getBlockNum LVC, which doesn't block
+	if latestBlockNum, err = getBlockNum(context.Background()); err != nil {
+		return false, err
+	}
+
+	blockNum, err := decodeBlockInput(tag)
+	if err != nil {
+		return false, err
+	}
+	return blockNum+uint64(BLOCK_NUM_ARCHIVE_HEIGHT) < latestBlockNum, nil
```

Comment:
Should this be `>`?

Reply:
hmm no. We return true (i.e. it's an archive request) if `blockNum` is too old/ too far behind.